packages/desktop-client/src/hooks/useLocalPref.ts

```diff
@@ -1,27 +1,43 @@
-import { useCallback } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+
+import { useLocalStorage } from 'usehooks-ts';
 
-import { savePrefs } from 'loot-core/src/client/actions';
-import { type State } from 'loot-core/src/client/state-types';
 import { type LocalPrefs } from 'loot-core/src/types/prefs';
 
+import { useMetadataPref } from './useMetadataPref';
+
 type SetLocalPrefAction<K extends keyof LocalPrefs> = (
   value: LocalPrefs[K],
 ) => void;
 
+/**
+ * Local preferences are scoped to a specific budget file.
+ */
 export function useLocalPref<K extends keyof LocalPrefs>(
   prefName: K,
 ): [LocalPrefs[K], SetLocalPrefAction<K>] {
-  const dispatch = useDispatch();
-  const setLocalPref = useCallback<SetLocalPrefAction<K>>(
-    value => {
-      dispatch(savePrefs({ [prefName]: value } as LocalPrefs));
+  const [budgetId] = useMetadataPref('id');
+
+  const [value, setValue] = useLocalStorage<LocalPrefs[K]>(
+    `${budgetId}-${prefName}`,
+    undefined,
+    {
+      deserializer: JSON.parse,
+      serializer: JSON.stringify,
     },
-    [prefName, dispatch],
-  );
-  const localPref = useSelector(
-    (state: State) => state.prefs.local?.[prefName] as LocalPrefs[K],
   );
 
-  return [localPref, setLocalPref];
+  // Migrate from old pref storage location (metadata.json) to local storage
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [metadataPref, setMetadataPref] = useMetadataPref(prefName as any);
+  useEffect(() => {
+    if (value !== undefined || metadataPref === undefined) {
+      return;
+    }
+
+    setValue(metadataPref);
+    setMetadataPref(undefined);
```

Comment:
This will wipe the metadata value which will wipe out settings if the user rolls back. I think this is also causing the metadata.json issue I mentioned in the separate comment.

I'm not sure why just setting it to undefined messes up the JSON tho - that's a weird one. 

Removing the line fixed it for me.

Reply:
Makes sense. Removed `setMetadataPref`: https://github.com/actualbudget/actual/pull/3458/commits/d6708a26596894453d1bb9460a35ee88b5296a4b